tools/flaky_test_monitor/process_results.go

```diff
@@ -107,7 +117,10 @@ func processTestRun(resultReader ResultReader) TestRun {
 	postProcessTestRun(packageResultMap)
 
 	testRun := finalizeTestRun(packageResultMap)
-	testRun.save(resultReader.getResultsFileName())
+
+	if resultReader.saveFiles() {
+		testRun.save(resultReader.getResultsFileName())
+	}
```

Comment:
The right way to solve this problem is to not have the file saving step inside `processTestRun` at all. writing to the file should be part of the `main` function.

Reply:
Updated - moved `save()` to `main()` and added conditional save logic to unit test